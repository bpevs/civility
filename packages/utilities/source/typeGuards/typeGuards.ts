// Types
export type Empty = EmptyArray | EmptyObj;
export type EmptyArray = never[];
export type Func = (...args: any[]) => any;
export type Nil = null | undefined;

// Interfaces
export interface EmptyObj {
  [index: string]: never;
  [index: number]: never;
}
export interface Obj {
  [key: string]: any;
  [key: number]: any;
}


// Type-guards
export function isArray(item: any): item is any[] {
  return Array.isArray(item)
}

export function isBoolean(item: any): item is boolean {
  return typeof item === "boolean"
}

export function isDate(item: any): item is Date {
  return item instanceof Date
}

export function isEmpty(item: any): item is EmptyArray | EmptyObj {
  return isNil(item) || !Object.keys(item).length
}

export function isError(item: any): item is Error {
  return item instanceof Error
}

export function isFunction(item: any): item is Func {
  return typeof item === "function"
}

export function isNil(item: any): item is Nil {
  return item == null
}

export function isNull(item: any): item is null {
  return item === null
}

export function isNumber(item: any): item is number {
  return typeof item === "number"
}

export function isObject(item: any): item is Obj {
  if (isNull(item)) return false
  return isFunction(item) || typeof item === "object"
}

export function isRegExp(item: any): item is RegExp {
  return item instanceof RegExp
}

export function isString(item: any): item is string {
  return typeof item === "string"
}

export function isUndefined(item: any): item is undefined {
  return item === undefined
}
