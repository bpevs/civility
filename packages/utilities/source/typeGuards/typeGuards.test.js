import { forEach } from "lodash"
import * as typeGuards from "./typeGuards"


const items = [
  "s", 0, 1, "1", Infinity, null, undefined, [], [ 1 ], {},
  { 0: 1, 1: 2 }, Number, new RegExp(/fsdf/), /fsdf/, 1.5234, Math.pow(2, 53),
  new Error("hello"), NaN, new Date("1/1/1")
];

forEach(typeGuards, check => {
  test(check.name, () => {
    items.forEach(item => {
      expect(`${check.name}(${item}): ${check(item)}`).toMatchSnapshot()
    })
  })
})
