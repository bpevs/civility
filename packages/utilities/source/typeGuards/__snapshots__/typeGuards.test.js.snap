// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isArray 1`] = `"isArray(s): false"`;

exports[`isArray 2`] = `"isArray(0): false"`;

exports[`isArray 3`] = `"isArray(1): false"`;

exports[`isArray 4`] = `"isArray(1): false"`;

exports[`isArray 5`] = `"isArray(Infinity): false"`;

exports[`isArray 6`] = `"isArray(null): false"`;

exports[`isArray 7`] = `"isArray(undefined): false"`;

exports[`isArray 8`] = `"isArray(): true"`;

exports[`isArray 9`] = `"isArray(1): true"`;

exports[`isArray 10`] = `"isArray([object Object]): false"`;

exports[`isArray 11`] = `"isArray([object Object]): false"`;

exports[`isArray 12`] = `"isArray(function Number() { [native code] }): false"`;

exports[`isArray 13`] = `"isArray(/fsdf/): false"`;

exports[`isArray 14`] = `"isArray(/fsdf/): false"`;

exports[`isArray 15`] = `"isArray(1.5234): false"`;

exports[`isArray 16`] = `"isArray(9007199254740992): false"`;

exports[`isArray 17`] = `"isArray(Error: hello): false"`;

exports[`isArray 18`] = `"isArray(NaN): false"`;

exports[`isArray 19`] = `"isArray(-2174774400000): false"`;

exports[`isArray 20`] = `"isArray([object Promise]): false"`;

exports[`isArray 21`] = `"isArray([object Set]): false"`;

exports[`isArray 22`] = `"isArray([object Map]): false"`;

exports[`isBoolean 1`] = `"isBoolean(s): false"`;

exports[`isBoolean 2`] = `"isBoolean(0): false"`;

exports[`isBoolean 3`] = `"isBoolean(1): false"`;

exports[`isBoolean 4`] = `"isBoolean(1): false"`;

exports[`isBoolean 5`] = `"isBoolean(Infinity): false"`;

exports[`isBoolean 6`] = `"isBoolean(null): false"`;

exports[`isBoolean 7`] = `"isBoolean(undefined): false"`;

exports[`isBoolean 8`] = `"isBoolean(): false"`;

exports[`isBoolean 9`] = `"isBoolean(1): false"`;

exports[`isBoolean 10`] = `"isBoolean([object Object]): false"`;

exports[`isBoolean 11`] = `"isBoolean([object Object]): false"`;

exports[`isBoolean 12`] = `"isBoolean(function Number() { [native code] }): false"`;

exports[`isBoolean 13`] = `"isBoolean(/fsdf/): false"`;

exports[`isBoolean 14`] = `"isBoolean(/fsdf/): false"`;

exports[`isBoolean 15`] = `"isBoolean(1.5234): false"`;

exports[`isBoolean 16`] = `"isBoolean(9007199254740992): false"`;

exports[`isBoolean 17`] = `"isBoolean(Error: hello): false"`;

exports[`isBoolean 18`] = `"isBoolean(NaN): false"`;

exports[`isBoolean 19`] = `"isBoolean(-2174774400000): false"`;

exports[`isBoolean 20`] = `"isBoolean([object Promise]): false"`;

exports[`isBoolean 21`] = `"isBoolean([object Set]): false"`;

exports[`isBoolean 22`] = `"isBoolean([object Map]): false"`;

exports[`isClient 1`] = `"isClient(s): true"`;

exports[`isClient 2`] = `"isClient(0): true"`;

exports[`isClient 3`] = `"isClient(1): true"`;

exports[`isClient 4`] = `"isClient(1): true"`;

exports[`isClient 5`] = `"isClient(Infinity): true"`;

exports[`isClient 6`] = `"isClient(null): true"`;

exports[`isClient 7`] = `"isClient(undefined): true"`;

exports[`isClient 8`] = `"isClient(): true"`;

exports[`isClient 9`] = `"isClient(1): true"`;

exports[`isClient 10`] = `"isClient([object Object]): true"`;

exports[`isClient 11`] = `"isClient([object Object]): true"`;

exports[`isClient 12`] = `"isClient(function Number() { [native code] }): true"`;

exports[`isClient 13`] = `"isClient(/fsdf/): true"`;

exports[`isClient 14`] = `"isClient(/fsdf/): true"`;

exports[`isClient 15`] = `"isClient(1.5234): true"`;

exports[`isClient 16`] = `"isClient(9007199254740992): true"`;

exports[`isClient 17`] = `"isClient(Error: hello): true"`;

exports[`isClient 18`] = `"isClient(NaN): true"`;

exports[`isClient 19`] = `"isClient(-2174774400000): true"`;

exports[`isClient 20`] = `"isClient([object Promise]): true"`;

exports[`isClient 21`] = `"isClient([object Set]): true"`;

exports[`isClient 22`] = `"isClient([object Map]): true"`;

exports[`isDate 1`] = `"isDate(s): false"`;

exports[`isDate 2`] = `"isDate(0): false"`;

exports[`isDate 3`] = `"isDate(1): false"`;

exports[`isDate 4`] = `"isDate(1): false"`;

exports[`isDate 5`] = `"isDate(Infinity): false"`;

exports[`isDate 6`] = `"isDate(null): false"`;

exports[`isDate 7`] = `"isDate(undefined): false"`;

exports[`isDate 8`] = `"isDate(): false"`;

exports[`isDate 9`] = `"isDate(1): false"`;

exports[`isDate 10`] = `"isDate([object Object]): false"`;

exports[`isDate 11`] = `"isDate([object Object]): false"`;

exports[`isDate 12`] = `"isDate(function Number() { [native code] }): false"`;

exports[`isDate 13`] = `"isDate(/fsdf/): false"`;

exports[`isDate 14`] = `"isDate(/fsdf/): false"`;

exports[`isDate 15`] = `"isDate(1.5234): false"`;

exports[`isDate 16`] = `"isDate(9007199254740992): false"`;

exports[`isDate 17`] = `"isDate(Error: hello): false"`;

exports[`isDate 18`] = `"isDate(NaN): false"`;

exports[`isDate 19`] = `"isDate(-2174774400000): false"`;

exports[`isDate 20`] = `"isDate([object Promise]): false"`;

exports[`isDate 21`] = `"isDate([object Set]): false"`;

exports[`isDate 22`] = `"isDate([object Map]): false"`;

exports[`isEmpty 1`] = `"isEmpty(s): false"`;

exports[`isEmpty 2`] = `"isEmpty(0): true"`;

exports[`isEmpty 3`] = `"isEmpty(1): true"`;

exports[`isEmpty 4`] = `"isEmpty(1): false"`;

exports[`isEmpty 5`] = `"isEmpty(Infinity): true"`;

exports[`isEmpty 6`] = `"isEmpty(null): true"`;

exports[`isEmpty 7`] = `"isEmpty(undefined): true"`;

exports[`isEmpty 8`] = `"isEmpty(): true"`;

exports[`isEmpty 9`] = `"isEmpty(1): false"`;

exports[`isEmpty 10`] = `"isEmpty([object Object]): true"`;

exports[`isEmpty 11`] = `"isEmpty([object Object]): false"`;

exports[`isEmpty 12`] = `"isEmpty(function Number() { [native code] }): true"`;

exports[`isEmpty 13`] = `"isEmpty(/fsdf/): true"`;

exports[`isEmpty 14`] = `"isEmpty(/fsdf/): true"`;

exports[`isEmpty 15`] = `"isEmpty(1.5234): true"`;

exports[`isEmpty 16`] = `"isEmpty(9007199254740992): true"`;

exports[`isEmpty 17`] = `"isEmpty(Error: hello): true"`;

exports[`isEmpty 18`] = `"isEmpty(NaN): true"`;

exports[`isEmpty 19`] = `"isEmpty(-2174774400000): true"`;

exports[`isEmpty 20`] = `"isEmpty([object Promise]): true"`;

exports[`isEmpty 21`] = `"isEmpty([object Set]): true"`;

exports[`isEmpty 22`] = `"isEmpty([object Map]): true"`;

exports[`isError 1`] = `"isError(s): false"`;

exports[`isError 2`] = `"isError(0): false"`;

exports[`isError 3`] = `"isError(1): false"`;

exports[`isError 4`] = `"isError(1): false"`;

exports[`isError 5`] = `"isError(Infinity): false"`;

exports[`isError 6`] = `"isError(null): false"`;

exports[`isError 7`] = `"isError(undefined): false"`;

exports[`isError 8`] = `"isError(): false"`;

exports[`isError 9`] = `"isError(1): false"`;

exports[`isError 10`] = `"isError([object Object]): false"`;

exports[`isError 11`] = `"isError([object Object]): false"`;

exports[`isError 12`] = `"isError(function Number() { [native code] }): false"`;

exports[`isError 13`] = `"isError(/fsdf/): false"`;

exports[`isError 14`] = `"isError(/fsdf/): false"`;

exports[`isError 15`] = `"isError(1.5234): false"`;

exports[`isError 16`] = `"isError(9007199254740992): false"`;

exports[`isError 17`] = `"isError(Error: hello): true"`;

exports[`isError 18`] = `"isError(NaN): false"`;

exports[`isError 19`] = `"isError(-2174774400000): false"`;

exports[`isError 20`] = `"isError([object Promise]): false"`;

exports[`isError 21`] = `"isError([object Set]): false"`;

exports[`isError 22`] = `"isError([object Map]): false"`;

exports[`isFunction 1`] = `"isFunction(s): false"`;

exports[`isFunction 2`] = `"isFunction(0): false"`;

exports[`isFunction 3`] = `"isFunction(1): false"`;

exports[`isFunction 4`] = `"isFunction(1): false"`;

exports[`isFunction 5`] = `"isFunction(Infinity): false"`;

exports[`isFunction 6`] = `"isFunction(null): false"`;

exports[`isFunction 7`] = `"isFunction(undefined): false"`;

exports[`isFunction 8`] = `"isFunction(): false"`;

exports[`isFunction 9`] = `"isFunction(1): false"`;

exports[`isFunction 10`] = `"isFunction([object Object]): false"`;

exports[`isFunction 11`] = `"isFunction([object Object]): false"`;

exports[`isFunction 12`] = `"isFunction(function Number() { [native code] }): true"`;

exports[`isFunction 13`] = `"isFunction(/fsdf/): false"`;

exports[`isFunction 14`] = `"isFunction(/fsdf/): false"`;

exports[`isFunction 15`] = `"isFunction(1.5234): false"`;

exports[`isFunction 16`] = `"isFunction(9007199254740992): false"`;

exports[`isFunction 17`] = `"isFunction(Error: hello): false"`;

exports[`isFunction 18`] = `"isFunction(NaN): false"`;

exports[`isFunction 19`] = `"isFunction(-2174774400000): false"`;

exports[`isFunction 20`] = `"isFunction([object Promise]): false"`;

exports[`isFunction 21`] = `"isFunction([object Set]): false"`;

exports[`isFunction 22`] = `"isFunction([object Map]): false"`;

exports[`isMap 1`] = `"isMap(s): false"`;

exports[`isMap 2`] = `"isMap(0): false"`;

exports[`isMap 3`] = `"isMap(1): false"`;

exports[`isMap 4`] = `"isMap(1): false"`;

exports[`isMap 5`] = `"isMap(Infinity): false"`;

exports[`isMap 6`] = `"isMap(null): false"`;

exports[`isMap 7`] = `"isMap(undefined): false"`;

exports[`isMap 8`] = `"isMap(): false"`;

exports[`isMap 9`] = `"isMap(1): false"`;

exports[`isMap 10`] = `"isMap([object Object]): false"`;

exports[`isMap 11`] = `"isMap([object Object]): false"`;

exports[`isMap 12`] = `"isMap(function Number() { [native code] }): false"`;

exports[`isMap 13`] = `"isMap(/fsdf/): false"`;

exports[`isMap 14`] = `"isMap(/fsdf/): false"`;

exports[`isMap 15`] = `"isMap(1.5234): false"`;

exports[`isMap 16`] = `"isMap(9007199254740992): false"`;

exports[`isMap 17`] = `"isMap(Error: hello): false"`;

exports[`isMap 18`] = `"isMap(NaN): false"`;

exports[`isMap 19`] = `"isMap(-2174774400000): false"`;

exports[`isMap 20`] = `"isMap([object Promise]): false"`;

exports[`isMap 21`] = `"isMap([object Set]): false"`;

exports[`isMap 22`] = `"isMap([object Map]): true"`;

exports[`isNil 1`] = `"isNil(s): false"`;

exports[`isNil 2`] = `"isNil(0): false"`;

exports[`isNil 3`] = `"isNil(1): false"`;

exports[`isNil 4`] = `"isNil(1): false"`;

exports[`isNil 5`] = `"isNil(Infinity): false"`;

exports[`isNil 6`] = `"isNil(null): true"`;

exports[`isNil 7`] = `"isNil(undefined): true"`;

exports[`isNil 8`] = `"isNil(): false"`;

exports[`isNil 9`] = `"isNil(1): false"`;

exports[`isNil 10`] = `"isNil([object Object]): false"`;

exports[`isNil 11`] = `"isNil([object Object]): false"`;

exports[`isNil 12`] = `"isNil(function Number() { [native code] }): false"`;

exports[`isNil 13`] = `"isNil(/fsdf/): false"`;

exports[`isNil 14`] = `"isNil(/fsdf/): false"`;

exports[`isNil 15`] = `"isNil(1.5234): false"`;

exports[`isNil 16`] = `"isNil(9007199254740992): false"`;

exports[`isNil 17`] = `"isNil(Error: hello): false"`;

exports[`isNil 18`] = `"isNil(NaN): false"`;

exports[`isNil 19`] = `"isNil(-2174774400000): false"`;

exports[`isNil 20`] = `"isNil([object Promise]): false"`;

exports[`isNil 21`] = `"isNil([object Set]): false"`;

exports[`isNil 22`] = `"isNil([object Map]): false"`;

exports[`isNull 1`] = `"isNull(s): false"`;

exports[`isNull 2`] = `"isNull(0): false"`;

exports[`isNull 3`] = `"isNull(1): false"`;

exports[`isNull 4`] = `"isNull(1): false"`;

exports[`isNull 5`] = `"isNull(Infinity): false"`;

exports[`isNull 6`] = `"isNull(null): true"`;

exports[`isNull 7`] = `"isNull(undefined): false"`;

exports[`isNull 8`] = `"isNull(): false"`;

exports[`isNull 9`] = `"isNull(1): false"`;

exports[`isNull 10`] = `"isNull([object Object]): false"`;

exports[`isNull 11`] = `"isNull([object Object]): false"`;

exports[`isNull 12`] = `"isNull(function Number() { [native code] }): false"`;

exports[`isNull 13`] = `"isNull(/fsdf/): false"`;

exports[`isNull 14`] = `"isNull(/fsdf/): false"`;

exports[`isNull 15`] = `"isNull(1.5234): false"`;

exports[`isNull 16`] = `"isNull(9007199254740992): false"`;

exports[`isNull 17`] = `"isNull(Error: hello): false"`;

exports[`isNull 18`] = `"isNull(NaN): false"`;

exports[`isNull 19`] = `"isNull(-2174774400000): false"`;

exports[`isNull 20`] = `"isNull([object Promise]): false"`;

exports[`isNull 21`] = `"isNull([object Set]): false"`;

exports[`isNull 22`] = `"isNull([object Map]): false"`;

exports[`isNumber 1`] = `"isNumber(s): false"`;

exports[`isNumber 2`] = `"isNumber(0): true"`;

exports[`isNumber 3`] = `"isNumber(1): true"`;

exports[`isNumber 4`] = `"isNumber(1): false"`;

exports[`isNumber 5`] = `"isNumber(Infinity): true"`;

exports[`isNumber 6`] = `"isNumber(null): false"`;

exports[`isNumber 7`] = `"isNumber(undefined): false"`;

exports[`isNumber 8`] = `"isNumber(): false"`;

exports[`isNumber 9`] = `"isNumber(1): false"`;

exports[`isNumber 10`] = `"isNumber([object Object]): false"`;

exports[`isNumber 11`] = `"isNumber([object Object]): false"`;

exports[`isNumber 12`] = `"isNumber(function Number() { [native code] }): false"`;

exports[`isNumber 13`] = `"isNumber(/fsdf/): false"`;

exports[`isNumber 14`] = `"isNumber(/fsdf/): false"`;

exports[`isNumber 15`] = `"isNumber(1.5234): true"`;

exports[`isNumber 16`] = `"isNumber(9007199254740992): true"`;

exports[`isNumber 17`] = `"isNumber(Error: hello): false"`;

exports[`isNumber 18`] = `"isNumber(NaN): true"`;

exports[`isNumber 19`] = `"isNumber(-2174774400000): true"`;

exports[`isNumber 20`] = `"isNumber([object Promise]): false"`;

exports[`isNumber 21`] = `"isNumber([object Set]): false"`;

exports[`isNumber 22`] = `"isNumber([object Map]): false"`;

exports[`isObject 1`] = `"isObject(s): false"`;

exports[`isObject 2`] = `"isObject(0): false"`;

exports[`isObject 3`] = `"isObject(1): false"`;

exports[`isObject 4`] = `"isObject(1): false"`;

exports[`isObject 5`] = `"isObject(Infinity): false"`;

exports[`isObject 6`] = `"isObject(null): false"`;

exports[`isObject 7`] = `"isObject(undefined): false"`;

exports[`isObject 8`] = `"isObject(): true"`;

exports[`isObject 9`] = `"isObject(1): true"`;

exports[`isObject 10`] = `"isObject([object Object]): true"`;

exports[`isObject 11`] = `"isObject([object Object]): true"`;

exports[`isObject 12`] = `"isObject(function Number() { [native code] }): true"`;

exports[`isObject 13`] = `"isObject(/fsdf/): true"`;

exports[`isObject 14`] = `"isObject(/fsdf/): true"`;

exports[`isObject 15`] = `"isObject(1.5234): false"`;

exports[`isObject 16`] = `"isObject(9007199254740992): false"`;

exports[`isObject 17`] = `"isObject(Error: hello): true"`;

exports[`isObject 18`] = `"isObject(NaN): false"`;

exports[`isObject 19`] = `"isObject(-2174774400000): false"`;

exports[`isObject 20`] = `"isObject([object Promise]): true"`;

exports[`isObject 21`] = `"isObject([object Set]): true"`;

exports[`isObject 22`] = `"isObject([object Map]): true"`;

exports[`isPromise 1`] = `"isPromise(s): false"`;

exports[`isPromise 2`] = `"isPromise(0): false"`;

exports[`isPromise 3`] = `"isPromise(1): false"`;

exports[`isPromise 4`] = `"isPromise(1): false"`;

exports[`isPromise 5`] = `"isPromise(Infinity): false"`;

exports[`isPromise 6`] = `"isPromise(null): false"`;

exports[`isPromise 7`] = `"isPromise(undefined): false"`;

exports[`isPromise 8`] = `"isPromise(): false"`;

exports[`isPromise 9`] = `"isPromise(1): false"`;

exports[`isPromise 10`] = `"isPromise([object Object]): false"`;

exports[`isPromise 11`] = `"isPromise([object Object]): false"`;

exports[`isPromise 12`] = `"isPromise(function Number() { [native code] }): false"`;

exports[`isPromise 13`] = `"isPromise(/fsdf/): false"`;

exports[`isPromise 14`] = `"isPromise(/fsdf/): false"`;

exports[`isPromise 15`] = `"isPromise(1.5234): false"`;

exports[`isPromise 16`] = `"isPromise(9007199254740992): false"`;

exports[`isPromise 17`] = `"isPromise(Error: hello): false"`;

exports[`isPromise 18`] = `"isPromise(NaN): false"`;

exports[`isPromise 19`] = `"isPromise(-2174774400000): false"`;

exports[`isPromise 20`] = `"isPromise([object Promise]): true"`;

exports[`isPromise 21`] = `"isPromise([object Set]): false"`;

exports[`isPromise 22`] = `"isPromise([object Map]): false"`;

exports[`isRegExp 1`] = `"isRegExp(s): false"`;

exports[`isRegExp 2`] = `"isRegExp(0): false"`;

exports[`isRegExp 3`] = `"isRegExp(1): false"`;

exports[`isRegExp 4`] = `"isRegExp(1): false"`;

exports[`isRegExp 5`] = `"isRegExp(Infinity): false"`;

exports[`isRegExp 6`] = `"isRegExp(null): false"`;

exports[`isRegExp 7`] = `"isRegExp(undefined): false"`;

exports[`isRegExp 8`] = `"isRegExp(): false"`;

exports[`isRegExp 9`] = `"isRegExp(1): false"`;

exports[`isRegExp 10`] = `"isRegExp([object Object]): false"`;

exports[`isRegExp 11`] = `"isRegExp([object Object]): false"`;

exports[`isRegExp 12`] = `"isRegExp(function Number() { [native code] }): false"`;

exports[`isRegExp 13`] = `"isRegExp(/fsdf/): true"`;

exports[`isRegExp 14`] = `"isRegExp(/fsdf/): true"`;

exports[`isRegExp 15`] = `"isRegExp(1.5234): false"`;

exports[`isRegExp 16`] = `"isRegExp(9007199254740992): false"`;

exports[`isRegExp 17`] = `"isRegExp(Error: hello): false"`;

exports[`isRegExp 18`] = `"isRegExp(NaN): false"`;

exports[`isRegExp 19`] = `"isRegExp(-2174774400000): false"`;

exports[`isRegExp 20`] = `"isRegExp([object Promise]): false"`;

exports[`isRegExp 21`] = `"isRegExp([object Set]): false"`;

exports[`isRegExp 22`] = `"isRegExp([object Map]): false"`;

exports[`isSet 1`] = `"isSet(s): false"`;

exports[`isSet 2`] = `"isSet(0): false"`;

exports[`isSet 3`] = `"isSet(1): false"`;

exports[`isSet 4`] = `"isSet(1): false"`;

exports[`isSet 5`] = `"isSet(Infinity): false"`;

exports[`isSet 6`] = `"isSet(null): false"`;

exports[`isSet 7`] = `"isSet(undefined): false"`;

exports[`isSet 8`] = `"isSet(): false"`;

exports[`isSet 9`] = `"isSet(1): false"`;

exports[`isSet 10`] = `"isSet([object Object]): false"`;

exports[`isSet 11`] = `"isSet([object Object]): false"`;

exports[`isSet 12`] = `"isSet(function Number() { [native code] }): false"`;

exports[`isSet 13`] = `"isSet(/fsdf/): false"`;

exports[`isSet 14`] = `"isSet(/fsdf/): false"`;

exports[`isSet 15`] = `"isSet(1.5234): false"`;

exports[`isSet 16`] = `"isSet(9007199254740992): false"`;

exports[`isSet 17`] = `"isSet(Error: hello): false"`;

exports[`isSet 18`] = `"isSet(NaN): false"`;

exports[`isSet 19`] = `"isSet(-2174774400000): false"`;

exports[`isSet 20`] = `"isSet([object Promise]): false"`;

exports[`isSet 21`] = `"isSet([object Set]): true"`;

exports[`isSet 22`] = `"isSet([object Map]): false"`;

exports[`isString 1`] = `"isString(s): true"`;

exports[`isString 2`] = `"isString(0): false"`;

exports[`isString 3`] = `"isString(1): false"`;

exports[`isString 4`] = `"isString(1): true"`;

exports[`isString 5`] = `"isString(Infinity): false"`;

exports[`isString 6`] = `"isString(null): false"`;

exports[`isString 7`] = `"isString(undefined): false"`;

exports[`isString 8`] = `"isString(): false"`;

exports[`isString 9`] = `"isString(1): false"`;

exports[`isString 10`] = `"isString([object Object]): false"`;

exports[`isString 11`] = `"isString([object Object]): false"`;

exports[`isString 12`] = `"isString(function Number() { [native code] }): false"`;

exports[`isString 13`] = `"isString(/fsdf/): false"`;

exports[`isString 14`] = `"isString(/fsdf/): false"`;

exports[`isString 15`] = `"isString(1.5234): false"`;

exports[`isString 16`] = `"isString(9007199254740992): false"`;

exports[`isString 17`] = `"isString(Error: hello): false"`;

exports[`isString 18`] = `"isString(NaN): false"`;

exports[`isString 19`] = `"isString(-2174774400000): false"`;

exports[`isString 20`] = `"isString([object Promise]): false"`;

exports[`isString 21`] = `"isString([object Set]): false"`;

exports[`isString 22`] = `"isString([object Map]): false"`;

exports[`isUndefined 1`] = `"isUndefined(s): false"`;

exports[`isUndefined 2`] = `"isUndefined(0): false"`;

exports[`isUndefined 3`] = `"isUndefined(1): false"`;

exports[`isUndefined 4`] = `"isUndefined(1): false"`;

exports[`isUndefined 5`] = `"isUndefined(Infinity): false"`;

exports[`isUndefined 6`] = `"isUndefined(null): false"`;

exports[`isUndefined 7`] = `"isUndefined(undefined): true"`;

exports[`isUndefined 8`] = `"isUndefined(): false"`;

exports[`isUndefined 9`] = `"isUndefined(1): false"`;

exports[`isUndefined 10`] = `"isUndefined([object Object]): false"`;

exports[`isUndefined 11`] = `"isUndefined([object Object]): false"`;

exports[`isUndefined 12`] = `"isUndefined(function Number() { [native code] }): false"`;

exports[`isUndefined 13`] = `"isUndefined(/fsdf/): false"`;

exports[`isUndefined 14`] = `"isUndefined(/fsdf/): false"`;

exports[`isUndefined 15`] = `"isUndefined(1.5234): false"`;

exports[`isUndefined 16`] = `"isUndefined(9007199254740992): false"`;

exports[`isUndefined 17`] = `"isUndefined(Error: hello): false"`;

exports[`isUndefined 18`] = `"isUndefined(NaN): false"`;

exports[`isUndefined 19`] = `"isUndefined(-2174774400000): false"`;

exports[`isUndefined 20`] = `"isUndefined([object Promise]): false"`;

exports[`isUndefined 21`] = `"isUndefined([object Set]): false"`;

exports[`isUndefined 22`] = `"isUndefined([object Map]): false"`;
